### ISSUE TITLE
Unable to start server/steamVR

### ISSUE THREAD

**Issue by ggrillone** [polygraphene/ALVR#113]
It seems I am able to connect my headset, but then it says "server failed to start, please press start server"
And after clicking that it attempts to launch steamVR, but errors out and everything crashes

Here's the error in the dialog pop-up from ALVR:
```
NvEnc CreateEncoder failed. Code=8 NvEncoder::CreateEncoder : m_nvenc.nvEncInitializeEncoder(m_hEncoder, &m_initializeParams) returned error 8 at c:\src\alvr\alvr_server\nvencoder.cpp:271
```

Here's the error log file that was generated:
````
Exception: Unhandled Exception.
ExceptionCode=C0000005
Address=00007FF8D773BE66 (00007FF8D7720000 + 000000000001BE66)
OSVer: MajorVersion=10 MinorVersion=0 Build=17134
Module: 00007FF8D7720000
========== Startup Log ==========
[21:44:16.518 703] HmdDriverFactory IServerTrackedDeviceProvider_004 (IServerTrackedDeviceProvider_004)
[21:44:16.518 703] HmdDriverFactory server return
[21:44:16.518 703] Serial Number: ALVR-001
[21:44:16.518 703] Model Number: ALVR driver server
[21:44:16.518 703] Render Target: 2048 1024
[21:44:16.518 703] Seconds from Vsync to Photons: 0.005000
[21:44:16.518 703] Display Frequency: 60.000000
[21:44:16.518 703] IPD: 0.064000
[21:44:16.518 703] debugOptions: Log:0 FrameIndex:0 FrameOutput:0 CaptureOutput:0 UseKeyedMutex:1
[21:44:16.518 703] EncoderOptions: -preset ll_hq -rc cbr_ll_hq -fps 60 -bitrate 30M -maxbitrate 30M
[21:44:16.518 703] ControlSocket::Startup Successfully bound to 127.0.0.1:9944
[21:44:16.518 703] UdpSocket::BindSocket successfully bound to 0.0.0.0:9944
[21:44:16.518 703] UdpSocket::BindQueueSocket bound queue socket. port=64530
[21:44:16.518 703] UdpSocket::Startup success
[21:44:16.518 703] Startup: ALVR Server Driver 2.2.1-alpha
[21:44:16.518 703] CRemoteHmd successfully initialized.
[21:44:16.518 703] TrackedDeviceAdded(HMD) Ret=1 SerialNumber=ALVR-001
[21:44:16.518 703] CRemoteHmd Activate 0
[21:44:16.551 667] Using NVIDIA GeForce GTX 750 as primary graphics adapter.
[21:44:16.551 667] OSVer:MajorVersion=10 MinorVersion=0 Build=17134
[21:44:16.551 667] Initializing CNvEncoder. Width=2048 Height=1024 Format=268435456 (useNV12:0)
[21:44:16.589 654] NvEnc Encoder Parameters:
NV_ENC_INITIALIZE_PARAMS:
encodeGUID: hevc
presetGUID: ll_hq
encodeWidth: 2048
encodeHeight: 1024
darWidth: 2048
darHeight: 1024
frameRateNum: 60
frameRateDen: 1
enableEncodeAsync: 1
reportSliceOffsets: 0
enableSubFrameWrite: 0
enableExternalMEHints: 0
enableMEOnlyMode: 0
enableWeightedPrediction: 0
maxEncodeWidth: 2048
maxEncodeHeight: 1024
maxMEHintCountsPerBlock: 000000A9457FCC28
NV_ENC_CONFIG:
profile: (default)
gopLength: 4294967295
frameIntervalP: 1
monoChromeEncoding: 0
frameFieldMode: 1
mvPrecision: 3
NV_ENC_RC_PARAMS:
    rateControlMode: 0x8
    constQP: 28, 31, 25
    averageBitRate:  30000000
    maxBitRate:      30000000
    vbvBufferSize:   0
    vbvInitialDelay: 0
    enableMinQP: 0
    enableMaxQP: 0
    enableInitialRCQP: 0
    enableAQ: 0
    qpMapMode: disabled
    enableLookahead: 0
    disableIadapt: 0
    disableBadapt: 0
    enableTemporalAQ: 0
    zeroReorderDelay: 0
    enableNonRefP: 0
    strictGOPTarget: 0
    aqStrength: 0
    minQP: 0, 0, 0
    maxQP: 0, 0, 0
    initialRCQP: 0, 0, 0
    temporallayerIdxMask: 0
    temporalLayerQP: 0, 0, 0, 0, 0, 0, 0, 0
    targetQuality: 0
    lookaheadDepth: 0
NV_ENC_CODEC_CONFIG (HEVC):
    level: 0
    tier: 0
    minCUSize: 0
    maxCUSize: 0
    useConstrainedIntraPred: 0
    disableDeblockAcrossSliceBoundary: 0
    outputBufferingPeriodSEI: 0
    outputPictureTimingSEI: 0
    outputAUD: 0
    enableLTR: 0
    disableSPSPPS: 0
    repeatSPSPPS: 1
    enableIntraRefresh: 0
    chromaFormatIDC: 0
    pixelBitDepthMinus8: 0
    idrPeriod: 4294967295
    intraRefreshPeriod: 0
    intraRefreshCnt: 0
    maxNumRefFramesInDPB: 0
    ltrNumFrames: 0
    vpsId: 0
    spsId: 0
    ppsId: 0
    sliceMode: 0
    sliceModeData: 0
    maxTemporalLayersMinus1: 0
    NV_ENC_CONFIG_HEVC_VUI_PARAMETERS:
        overscanInfoPresentFlag: 0
        overscanInfo: 0
        videoSignalTypePresentFlag: 0
        videoFormat: 0
        videoFullRangeFlag: 0
        colourDescriptionPresentFlag: 0
        colourPrimaries: 0
        transferCharacteristics: 0
        colourMatrix: 0
        chromaSampleLocationFlag: 0
        chromaSampleLocationTop: 0
        chromaSampleLocationBot: 0
        bitstreamRestrictionFlag: 0
    ltrTrustMode: 0

[21:44:16.589 654] NvEnc CreateEncoder failed. Code=8 NvEncoder::CreateEncoder : m_nvenc.nvEncInitializeEncoder(m_hEncoder, &m_initializeParams) returned error 8 at c:\src\alvr\alvr_server\nvencoder.cpp:271

[21:44:17.974 788] Control Command: GetConfig 
[21:44:17.974 788] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:17.976 802] Control Command: GetRequests 
[21:44:17.976 802] SendCommandResponse: 
[21:44:18.975 934] Control Command: GetConfig 
[21:44:18.975 934] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:18.976 984] Control Command: GetRequests 
[21:44:18.976 984] SendCommandResponse: 
[21:44:19.975 059] Control Command: GetConfig 
[21:44:19.975 059] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:19.976 111] Control Command: GetRequests 
[21:44:19.976 111] SendCommandResponse: 
[21:44:20.352 965] Error on recv. close control client: 10054
[21:44:23.310 214] Control Command: GetConfig 
[21:44:23.310 214] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:23.316 210] Control Command: GetRequests 
[21:44:23.316 210] SendCommandResponse: 
[21:44:24.306 902] Control Command: GetConfig 
[21:44:24.306 902] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:24.307 890] Control Command: GetRequests 
[21:44:24.307 890] SendCommandResponse: 
[21:44:25.303 903] Control Command: GetConfig 
[21:44:25.303 903] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:25.304 642] Control Command: GetRequests 
[21:44:25.304 642] SendCommandResponse: 
[21:44:26.036 581] Received packet. Type=3
[21:44:26.036 581] Got recover connection message from 192.168.0.15:9944.
[21:44:26.036 581] Received packet. Type=1
[21:44:26.036 581] Hello Message: Oculus_Pacific Version=16 Hz=72
[21:44:26.303 821] Control Command: GetConfig 
[21:44:26.303 821] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:26.303 821] Control Command: GetRequests 
[21:44:26.303 821] SendCommandResponse: 192.168.0.15:9944 1 72 Oculus_Pacific

[21:44:26.808 724] Received packet. Type=1
[21:44:26.808 724] Hello Message: Oculus_Pacific Version=16 Hz=72
[21:44:27.303 288] Control Command: GetConfig 
[21:44:27.303 288] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:27.304 272] Control Command: GetRequests 
[21:44:27.304 272] SendCommandResponse: 192.168.0.15:9944 1 72 Oculus_Pacific

[21:44:27.815 885] Received packet. Type=1
[21:44:27.815 885] Hello Message: Oculus_Pacific Version=16 Hz=72
[21:44:28.302 303] Control Command: GetConfig 
[21:44:28.302 303] SendCommandResponse: Connected 0
Client 0.0.0.0:0
ClientName 
RefreshRate 1702128737
Streaming 0
debugLog 0
debugFrameIndex 0
debugFrameOutput 0
debugCaptureOutput 0
useKeyedMutex 1
controllerTriggerMode 24
controllerTrackpadClickMode 28
controllerTrackpadTouchMode 29
controllerRecenterButton 0
[21:44:28.302 303] Control Command: GetRequests 
[21:44:28.302 303] SendCommandResponse: 192.168.0.15:9944 1 72 Oculus_Pacific

[21:44:28.891 080] Received packet. Type=1
[21:44:28.891 080] Hello Message: Oculus_Pacific Version=16 Hz=72
[21:44:28.989 465] Control Command: Connect 192.168.0.15:9944
[21:44:28.990 464] Connected to 192.168.0.15:9944 refreshRate=72
[21:44:28.990 464] SendCommandResponse: OK

[21:44:29.161 370] Received packet. Type=5
[21:44:29.161 370] Stream control message: Start stream.
[21:44:29.161 370] Received packet. Type=6
[21:44:29.161 370] got tracking info 1095 -0.041530 0.026956 0.010495 0.998718
[21:44:29.161 370] Unhandled Exception.
ExceptionCode=C0000005
Address=00007FF8D773BE66 (00007FF8D7720000 + 000000000001BE66)
========== Tail Log 1 ==========
========== Tail Log 2 ==========
========== Exception info start ==========
SymGetSymFromAddr64: 0, SymGetLineFromAddr64: 0
ExceptionCode=C0000005 Address=00007FF8D773BE66
========== Exception info end ==========

````

**Comment by vrokolos** [comment]
750 doesn't support h265. use h264

**Comment by ggrillone** [comment]
could you clarify in layman's terms what you mean by h264? Thanks in advance

**Comment by ggrillone** [comment]
do I just need to use an older version of alvr?

**Comment by polygraphene** [comment]
You need to choose H.264 AVC on video tab.

**Comment by polygraphene** [comment]
H.265 and H.264 are video encoder name.
H.265 produces better quality video than H.264, but it is only supported on newer NVIDIA GPU.

**Comment by ggrillone** [comment]
thank you for the explanation :)
